src/Routing/Route/Route.php

```diff
@@ -694,11 +694,6 @@ public function match(array $url, array $context = []): ?string
         }
         unset($url['_method'], $url['[method]'], $defaults['_method']);
 
-        // Missing defaults is a fail.
-        if (array_diff_key($defaults, $url) !== []) {
```

Comment:
@markstory I don't quite understand the purpose of this check. If it's indeed necessary then we should have a test for it. Currently removing this check doesn't cause any test failure.

Reply:
Nevermind, this condition is also covered by the check on 703.